src/js/components/PostTweet.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { postTweet } from '../services/timelineService';
+
+export default class PostTweet extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            message: '',
+            error: null,
+            isLoaded: false,
+            charCount: 0
+        };
+    }
+
+    handleInputChange(event){
+        this.setState({
+            message: event.target.value,
+            charCount: event.target.value.length
+        });
+    }
+
+    postTweet(){
+        this.setState({
```

Comment:
If you're calling set state later within the same function, do you need to call it here? If it's necessary to make sure the component has a specific state before calling postTweet then it would be a good idea to call postTweet within the setState callback

Reply:
this is needed to toggle the 'isLoaded' state which controls whether there is a report or not

Reply:
Since you're updating isLoaded again later it's sort of pointless to set it initially at the beginning of this method. Please read this for reference: https://stackoverflow.com/questions/48563650/does-react-keep-the-order-for-state-updates/48610973#48610973
However, if you feel it is super necessary, postTweet should be moved into the setState callback